engine/src/flutter/lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -1013,6 +1022,117 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
+  ///
+  /// Updates [typographySettings] with the new value.
+  DomResizeObserver? _typographySettingsObserver;
+  DomElement? _typographyMeasurementElement;
+
+  /// Updates [typographySettings] and invokes [onPlatformConfigurationChanged] and
+  /// [onMetricsChanged] callbacks if [typographySettings] changed.
+  void _updateTypographySettings(ui.TypographySettings? value) {
+    if (configuration.typographySettings != value) {
+      configuration = configuration.copyWith(typographySettings: value);
+      invokeOnPlatformConfigurationChanged();
+      invokeOnMetricsChanged();
+    }
+  }
+
+  ui.TypographySettings? _computeTypographySettings() {
+    final double? lineHeight = parseStyleProperty(
+      _typographyMeasurementElement!,
+      'line-height',
+    )?.toDouble();
+    final double? wordSpacing = parseStyleProperty(
+      _typographyMeasurementElement!,
+      'word-spacing',
+    )?.toDouble();
+    final double? letterSpacing = parseStyleProperty(
+      _typographyMeasurementElement!,
+      'letter-spacing',
+    )?.toDouble();
+    // There is no direct CSS property for paragraph spacing,
+    // so on the web this feature is usually implemented
+    // by extension authors by leveraging `margin-bottom` on
+    // the `p` element.
+    final double? paragraphSpacing = parseStyleProperty(
+      _typographyMeasurementElement!,
+      'margin-bottom',
+    )?.toDouble();
+    if (lineHeight == null &&
+        wordSpacing == null &&
+        letterSpacing == null &&
+        paragraphSpacing == null) {
+      return null;
+    }
+    return ui.TypographySettings(
+      lineHeight: lineHeight,
+      letterSpacing: letterSpacing,
+      wordSpacing: wordSpacing,
+      paragraphSpacing: paragraphSpacing,
+    );
+  }
+
+  /// Set the callback function for updating [typographySettings] based on
+  /// the sizing changes of an off-screen element with text.
+  void _addTypographySettingsObserver() {
+    _typographyMeasurementElement = createDomHTMLParagraphElement();
+    _typographyMeasurementElement!.text = 'flutter typography measurement';
+    // The element should be hidden from screen readers.
+    _typographyMeasurementElement!.setAttribute('aria-hidden', 'true');
+    final DomCSSStyleDeclaration style = _typographyMeasurementElement!.style;
+    style.position = 'absolute';
+    // The element should be off-screen and not visible.
+    style.top = '-9999px';
+    style.left = '-9999px';
+    style.visibility = 'hidden';
+    // The element should be sensitive to letter-spacing, word-spacing,
+    // and line-height changes.
+    style.width = 'auto';
+    style.height = 'auto';
+    style.whiteSpace = 'nowrap';
+    // Set text spacing properties defaults.
+    const double spacingDefault = 100.0;
+    style.lineHeight = '${spacingDefault}px';
+    style.letterSpacing = '${spacingDefault}px';
+    style.wordSpacing = '${spacingDefault}px';
+    style.margin = '0px 0px ${spacingDefault}px 0px';
```

Comment:
Applying these default styles directly on the element, are browser extensions able to override them via global styles?

Reply:
Normally, inline styles that are set directly on the element (which is what you are doing here) will override any global styles. But extensions are using `!important` to force their styles to take precedence: https://github.com/actum/text-spacing-editor/blob/6782a09bfd55d5ad7791c498804ecfbf28ece0dc/helpers/buildCSSToInject.ts#L20-L30

So you are good 🙂